lib/utils.pm

```diff
@@ -40,10 +45,12 @@ sub wait_for_desktop {
         if (match_has_tag('openqa-desktop-locked')) {
             send_key 'esc';
             handle_gui_password;
+            last if (check_screen('generic-desktop', 10));
```

Comment:
Please use a multi-tag `assert_screen` with `match_has_tag` instead of `check_screen` with non-zero timeout to prevent introducing any timing dependant behaviour, to save test execution time as well as state more explicitly from the testers point of view what are the expected alternatives. For example:

```perl
assert_screen([qw(yast2_console-finished yast2_missing_package)]);
if (match_has_tag('yast2_missing_package')) {
    send_key 'alt-o';  # confirm package installation
    assert_screen 'yast2_console-finished';
}
```

See https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/CONTRIBUTING.md#coding-style for more details

Reply:
I did this exactly like this on purpose.
This is a hard requirement to fix race conditions when logging in, that would appear nearly always when I run this test locally.

The timeout is intended and wanted, and I only use a single needle because this is the needle that is expected after performing login and it won't introduce a big delay in the case of success.

The problem is that after login, for some second or so we still get the old screen content.

Reply:
not resolved

Reply:
The "wait_for_desktop" story has more context. There are already multiple tickets related to that and the near-duplicate code that we have in a similar manner in os-autoinst-distri-opensuse. Introducing more differences is not a good idea.

Reply:
@baierjan didn't you recently worked on related code and have some suggestion here?

Reply:
Yes, I did. If I remember correctly, I did not need to touch the code and my problem was solved with https://github.com/os-autoinst/os-autoinst-needles-openQA/pull/23.